File: backend/app/schemas/tailored_resume.py
from datetime import datetime

from pydantic import BaseModel, Field


class TailoredResumeItem(BaseModel):
    title: str | None = None
    bullets: list[str] = Field(default_factory=list)


class TailoredResumeSection(BaseModel):
    heading: str
    items: list[TailoredResumeItem] = Field(default_factory=list)


class TailoredResumeContent(BaseModel):
    summary: str | None = None
    sections: list[TailoredResumeSection] = Field(default_factory=list)
    notes: str | None = None


class JobTailoredResumeResult(TailoredResumeContent):
    job_id: int
    generated_at: datetime


class JobTailoredResumeRead(BaseModel):
    computed: bool
    resume: JobTailoredResumeResult | None = None

File: frontend/src/types/tailored_resume.ts
export type TailoredResumeItem = {
  title: string | null;
  bullets: string[];
};

export type TailoredResumeSection = {
  heading: string;
  items: TailoredResumeItem[];
};

export type JobTailoredResumeResult = {
  job_id: number;
  summary: string | null;
  sections: TailoredResumeSection[];
  notes: string | null;
  generated_at: string;
};

export type JobTailoredResumeRead = {
  computed: boolean;
  resume: JobTailoredResumeResult | null;
};

export function formatTailoredResumeDate(iso: string): string {
  return new Date(iso).toLocaleString(undefined, {
    dateStyle: "medium",
    timeStyle: "short",
  });
}

export function tailoredResumeToText(result: JobTailoredResumeResult): string {
  const lines: string[] = [];

  if (result.summary) {
    lines.push(result.summary, "");
  }

  for (const section of result.sections) {
    lines.push(section.heading.toUpperCase());
    for (const item of section.items) {
      if (item.title) {
        lines.push(item.title);
      }
      for (const bullet of item.bullets) {
        lines.push(`• ${bullet}`);
      }
      lines.push("");
    }
  }

  return lines.join("\n").trim();
}
